### PR TITLE
SOLR-10285: Skip LEADER messages when there are leader only shards

### PR DESCRIPTION
**Pull request by dragonsinth**
Tests passing

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import org.apache.solr.common.cloud.DocCollection;
+import org.apache.solr.common.cloud.Replica;
+import org.apache.solr.common.cloud.Slice;
+import org.junit.BeforeClass;
+
+public class TestShardsWithSingleReplica extends SolrCloudTestCase {
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    // Horrible copy-patsa.
```
Comment (jhump) [inline solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: pasta

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        }
+        return true;
+      }
+    });
+
+    CollectionAdminResponse resp2 = new CollectionAdminRequest.OverseerStatus().process(cluster.getSolrClient());
+    assertEquals(getNumLeaderOpeations(resp), getNumLeaderOpeations(resp2));
+  }
+
+  private int getNumLeaderOpeations(CollectionAdminResponse resp) {
```
Comment (jhump) [inline solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: typo in Operations

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import org.apache.solr.common.cloud.Slice;
+import org.junit.BeforeClass;
+
+public class TestShardsWithSingleReplica extends SolrCloudTestCase {
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    // Horrible copy-patsa.
+    System.setProperty("solr.test.sys.prop1", "propone");
+    System.setProperty("solr.test.sys.prop2", "proptwo");
```
Comment (jhump) [inline solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: what do these do?

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+          }
+          if (activeReplicas != 1)
+            return false;
+        }
+        return true;
+      }
+    });
+
+    CollectionAdminResponse resp2 = new CollectionAdminRequest.OverseerStatus().process(cluster.getSolrClient());
+    assertEquals(getNumLeaderOpeations(resp), getNumLeaderOpeations(resp2));
```
Comment (jhump) [inline solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: Assert they are both 2, one for each shard, instead?

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java (hunk 1 above)
Comment (dragonsinth) [reply solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: whoops

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java (hunk 3 above)
Comment (dragonsinth) [reply solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: I have no idea, but the solr config template blows up if these are set.

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java (hunk 4 above)
Comment (dragonsinth) [reply solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: This section is actually identical to what landed on master.

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java (hunk 2 above)
Comment (dragonsinth) [reply solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: typo is from apache/master

Comment (ianrose14) [reply solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: 😢

File: solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java (hunk 3 above)
Comment (ianrose14) [reply solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java]: 😆

**Pull request merged**